openapi/openapi.yml

```diff
@@ -0,0 +1,246 @@
+            format: int64
+      responses:
+        '400':
+          description: Invalid room value
+  /room/{roomId}/peer:
+    post:
+      tags:
+        - room
+      summary: Add peer
+      operationId: addPeer
+      requestBody:
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/PeerConfig'
+      responses:
+        '200':
+          description: successful operation
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Peer'
+        '500':
+          description: reached max number of peers
+  /room/{roomId}/consumer:
+    post:
+      tags:
+        - room
+      summary: Create consumer
+      operationId: createConsumer
+      requestBody:
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/ConsumerConfig'
+      responses:
+        '200':
+          description: successful operation
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Consumer'
+  /room/{roomId}/producer:
+    post:
+      tags:
+        - room
+      summary: Create producer
+      operationId: createProducer
+      requestBody:
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/ProducerConfig'
+      responses:
+        '200':
+          description: successful operation
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Producer'
+components:
+  schemas:
+    RoomConfig:
+      type: object
+      properties:
+        maxPeers:
+          type: integer
+          format: int64
+          example: 10
+    Room:
+      type: object
+      properties: 
+        id:
+          type: integer
+          format: int64
+        config:
+          type: object
+          schema: 
+              $ref: '#/components/schemas/RoomConfig'
+        producers:
+          type: array
+          items:
+            $ref: '#/components/schemas/Producer'
+        consumers:
+          type: array
+          items:
+            $ref: '#/components/schemas/Consumer'
+    ProducerConfig:
+      type: object
+      properties:
+        type:
+          $ref: '#/components/schemas/ProducerType'
+    Producer:
+      type: object
+      description: "
+        Producer is a server side entity that publishes a track.
+        Example producer is RTSP, which connects to a remote camera
+        or FileReader, which reads a track from a file. Producer 
+        doesn't subscribe to tracks and is spawned just after adding it.
+        "
+      properties:
+        id:
+          type: integer
+          format: int64
+        type:
+          $ref: '#/components/schemas/ProducerType'
+    ConsumerConfig:
+      type: object
+      properties: 
+        type:
+          $ref: '#/components/schemas/ConsumerType'
```

Comment:
I think we also need a space for endpoint-specific configuration, it's a very basic feature and I think we will actually need it for HLS for example.

Also, how about allowing to set custom id? This might be a nice feature for the business logic to get a simple "hls" as an id instead of 64, or whatever it is, random letters.